frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -52,86 +51,97 @@ const getIconClass = (status: ClusterUpdateStatus) => {
+  </div>
+  <div>
+    <Button bsStyle="primary" onClick={() => (clusterUpdateModal({cv}))}>
+      Update now
+    </Button>
+  </div>
+</React.Fragment>;
 
-const RetrieveUpdatesFailedAlert: React.SFC<RetrieveUpdatesFailedAlertProps> = ({condition}) => <div className="alert alert-danger">
-  <i className="pficon pficon-error-circle-o" aria-hidden="true" />
-  <strong>Could not retrieve updates.</strong>
-  {condition.message && ` ${condition.message}.`}
-</div>;
+const UpdatingMessage: React.SFC<CVStatusMessageProps> = ({cv, iconClass}) => {
+  const updatingCondition = getClusterVersionCondition(cv, ClusterVersionConditionType.Progressing, K8sResourceConditionStatus.True);
+  return <React.Fragment>
+    {updatingCondition.message && <div>{iconClass && <i className={iconClass} aria-hidden={true}></i>} {updatingCondition.message}</div>}
+    <Link to="/settings/cluster/clusteroperators">View detailed progress</Link>
+  </React.Fragment>;
+};
 
-const UpdateInProgressAlert: React.SFC<UpdateInProgressAlertProps> = ({condition}) => <div className="alert alert-info">
-  <i className="pficon pficon-info" aria-hidden={true} />
-  <strong>Cluster update in progress.</strong>
-  &nbsp;
-  {condition.message && `${condition.message}. `}
-  View <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">Cluster Operators</Link> for more details.
+const ErrorRetrievingMessage: React.SFC<StatusMessageProps> = ({iconClass}) => <div>
+  {iconClass && <i className={iconClass} aria-hidden={true}></i>} Error retrieving
 </div>;
 
-const UpdatesAvailableAlert: React.SFC<UpdatesAvailableAlertProps> = ({cv}) => {
-  const currentlyUpdating = isProgressing(cv) || updateFailing(cv);
-  const titleText = currentlyUpdating ? 'Other updates are available.' : 'Cluster update is available.';
-  const buttonText = currentlyUpdating ? 'Update to a different version.' : 'Update now.';
-  return <div className="alert alert-info">
-    <i className="pficon pficon-info" aria-hidden={true} />
-    <strong>
-      {titleText}
-    </strong>
-    <Button bsStyle="link" className="co-m-modal-link" onClick={()=> (clusterUpdateModal({cv}))}>
-      {buttonText}
-    </Button>
-  </div>;
-};
+const FailingMessage: React.SFC<StatusMessageProps> = ({iconClass}) => <React.Fragment>
+  <div>
+    {iconClass && <i className={iconClass} aria-hidden={true}></i>} Failing
+  </div>
+  <Link to="/settings/cluster/clusteroperators">View detailed progress</Link>
+</React.Fragment>;
 
+const UpToDateMessage: React.SFC<StatusMessageProps> = ({iconClass}) => <span>
+  {iconClass && <i className={iconClass} aria-hidden={true}></i>} Up to date
+</span>;
 
 const UpdateStatus: React.SFC<UpdateStatusProps> = ({cv}) => {
   const status = getClusterUpdateStatus(cv);
   const iconClass = getIconClass(status);
+  switch (status) {
+    case ClusterUpdateStatus.UpdatesAvailable:
+      return <UpdatesAvailableMessage cv={cv} iconClass={iconClass} />;
```

Comment:
You shouldn't have to pass the icon class into each component. Since it's a separate component for each status, they should already know what icon class to use.